File: utilcode/app/src/main/java/com/blankj/utilcode/app/UtilCodeApp.java
package com.blankj.utilcode.app;

import android.content.Context;

import com.blankj.lib.base.BaseApplication;
import com.blankj.utilcode.util.Utils;


/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/10/12
 *     desc  : app about utils
 * </pre>
 */
public class UtilCodeApp extends BaseApplication {

    private static UtilCodeApp sInstance;

    public static UtilCodeApp getInstance() {
        return sInstance;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
    }

    @Override
    public void onCreate() {
        Utils.init(this);
        super.onCreate();
        sInstance = this;
    }
}

File: utilcode/app/src/main/java/com/blankj/utilcode/app/UtilCodeApp.kt
package com.blankj.utilcode.app

import com.blankj.lib.base.BaseApplication
import com.blankj.utilcode.util.Utils


/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/10/12
 * desc  : app about utils
 * ```
 */
class UtilCodeApp : BaseApplication() {

    companion object {
        lateinit var instance: UtilCodeApp
            private set
    }

    override fun onCreate() {
        Utils.init(this)
        super.onCreate()
        instance = this
//        BusUtils.registerClient("com.blankj.androidutilcode")
    }
}


